react/index.tsx

```diff
@@ -6,54 +6,66 @@ const CSS_HANDLES = ['container'] as const
 
 enum Positions {
   BOTTOM = 'bottom',
+  TOP = 'top',
 }
 
 interface Props {
   position?: Positions
+  margin?: number
 }
 
 interface StorefrontComponent extends FC<Props> {
   schema?: any
 }
 
-const StickyLayoutComponent: StorefrontComponent = ({
-  children,
-  position,
-}) => {
+const StickyLayoutComponent: StorefrontComponent = ({ children, position, margin }) => {
+  const containerReference = useRef<HTMLDivElement>(null)
   const container = useRef<HTMLDivElement>(null)
   const handles = useCssHandles(CSS_HANDLES)
 
-  if (position !== Positions.BOTTOM) {
-    // Only 'bottom' position supported for now!
-    return <Fragment>{children}</Fragment>
-  }
-
-  const handleScroll = useCallback((e?: Event) => {
-    const target = (e && e.target) as HTMLDivElement | null
-    if (!container.current) {
+  const handleScroll = useCallback(() => {
+    if (!containerReference.current || !container.current) {
       return
     }
 
-    // We have to get this baseTop variable to handle cases when a modal changes the top style of the base element, resetting the window.pageYOfsset
-    const baseTop = parseInt(
-      path(['scrollingElement', 'style', 'top'], target) || '0',
-      10
-    )
-    const componentTop = container.current.offsetTop + baseTop
-
-    const newTop =
-      window.innerHeight -
-      componentTop -
-      container.current.clientHeight +
-      window.pageYOffset
-
-    if (newTop < 0) {
-      container.current.style.transform = `translate3d(0, ${newTop}px, 0)`
-      return
+    let stickyPosition = 0
+
+    let containerReferencePosition =
+      containerReference.current &&
+      containerReference.current.offsetTop
+
+    if (position === Positions.BOTTOM) {
+      containerReferencePosition += container.current.clientHeight
+      stickyPosition = window.pageYOffset + window.innerHeight
+    } else if (position === Positions.TOP) {
```

Comment:
What happens if position is `FOO`? Shouldn't we do something like we were doing before, short circuiting like:

```jsx
return <Fragment>{children}</Fragment>
```
(not at this point though otherwise it would break the rules of hooks)

Reply:
Done in https://github.com/vtex-apps/sticky-layout/pull/7/files#diff-5606612e96ad0dac35762b7b2d88d335R22.